raddar/crud/crud.py

```diff
@@ -1,68 +1,80 @@
+        **secret_to_create.dict(), analysis_id=analysis_id
+    )
+    return await database.execute(query=query)
 
 
-def create_project(db: Session, project: schemas.ProjectCreate):
-    db_project = models.Project(name=project.name)
-    db.add(db_project)
-    db.commit()
-    db.refresh(db_project)
-    return db_project
+async def create_project(project_to_create: schemas.ProjectBase):
+    query = project.insert(None).values(**project_to_create.dict())
+    return await database.execute(query=query)
 
 
-def create_analysis(
-    db: Session,
-    project: schemas.ProjectCreate,
-    branch_name: str,
+async def create_analysis(
+    project_id: int,
+    analysis_to_create: schemas.AnalysisBase,
     ref_name: str,
-    scan_origin: Literal["manual", "github-webhook"],
+    scan_origin: models.ScanOrigin,
+    secrets_to_create: List[schemas.SecretBase],
 ):
-    db_project = get_project_by_name(db, project_name=project.name)
-
-    if not db_project:
-        db_project = create_project(db, project)
-
-    db_analysis = models.Analysis(
-        execution_date=datetime.now(),
-        branch_name=branch_name,
+    now = datetime.now(timezone.utc)
+    query = analysis.insert(None).values(
+        execution_date=now,
+        branch_name=get_branch_name(analysis_to_create.branch_name),
         ref_name=ref_name,
         scan_origin=scan_origin,
-        project_id=db_project.id,
+        project_id=project_id,
     )
-    db.add(db_analysis)
-    db.commit()
-    db.refresh(db_analysis)
-    return db_analysis
+    analysis_returned_id = await database.execute(query=query)
 
+    secrets_returned = []
+    for secret_to_create in secrets_to_create:
+        secret_returned_id = await create_analysis_secret(
+            secret_to_create, analysis_returned_id
+        )
+        secrets_returned.append({**secret_to_create.dict(), "id": secret_returned_id})
 
-def get_project_by_name(db: Session, project_name: str):
-    return db.query(models.Project).filter(models.Project.name == project_name).first()
+    return {
+        **analysis_to_create.dict(),
+        "id": analysis_returned_id,
+        "execution_date": now,
+        "ref_name": ref_name,
+        "scan_origin": scan_origin,
+        "project_id": project_id,
+        "secrets": secrets_to_create,
+    }
 
 
-def get_analysis_by_name_and_ref(
-    db: Session, project_name: str, branch_name: str, ref_name: str
+async def get_project_analysis_by_name_and_ref(
+    project_name: str, branch_name: str, ref_name: str
 ):
-    return (
-        db.query(models.Analysis)
-        .filter(
-            models.Project.name == project_name,
-            or_(ref_name == ref_name, branch_name == branch_name),
+    query = project.select().where(
+        and_(
+            project.c.name == project_name,
+            or_(analysis.c.branch_name == branch_name, analysis.c.ref_name == ref_name),
         )
-        .order_by(models.Analysis.execution_date.desc())
-        .first()
```

Comment:
You removed the `order_by` and the `first`?

Reply:
refactor ^^